packages/twenty-front/src/modules/action-menu/actions/utils/getActionConfig.ts

```diff
@@ -2,26 +2,81 @@ import { DEFAULT_RECORD_ACTIONS_CONFIG } from '@/action-menu/actions/record-acti
 import { WORKFLOW_ACTIONS_CONFIG } from '@/action-menu/actions/record-actions/constants/WorkflowActionsConfig';
 import { WORKFLOW_RUNS_ACTIONS_CONFIG } from '@/action-menu/actions/record-actions/constants/WorkflowRunsActionsConfig';
 import { WORKFLOW_VERSIONS_ACTIONS_CONFIG } from '@/action-menu/actions/record-actions/constants/WorkflowVersionsActionsConfig';
+import { NoSelectionRecordActionKeys } from '@/action-menu/actions/record-actions/no-selection/types/NoSelectionRecordActionsKeys';
 import { ActionConfig } from '@/action-menu/actions/types/ActionConfig';
 import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
 import { ObjectMetadataItem } from '@/object-metadata/types/ObjectMetadataItem';
 import { isDefined } from 'twenty-shared/utils';
 
-export const getActionConfig = (
-  objectMetadataItem?: ObjectMetadataItem,
-): Record<string, ActionConfig> => {
+export const getActionConfig = ({
+  objectMetadataItem,
+  notePermissions,
+  taskPermissions,
+  companyPermissions,
+  personPermissions,
+  opportunityPermissions,
+  workflowPermissions,
+}: {
+  objectMetadataItem?: ObjectMetadataItem;
+  notePermissions: { canRead: boolean };
```

Comment:
does not look to scale well, shouldn't we pass a objectPermissions rather? which can be dynamic?